tests/ct_connection.c

```diff
@@ -320,17 +404,55 @@ void test_async_multicluster() {
     event_base_free(base);
 }
 
+// Connect to a cluster and authenticate using username and password
+void test_async_username_ok() {
+    // Prepare by adding a user to all instances in the cluster
+    create_user_in_cluster(CLUSTER_NODE);
+
+    // Connect to the cluster using username and password
+    redisClusterAsyncContext *acc = redisClusterAsyncContextInit();
+    assert(acc);
+    redisClusterAsyncSetConnectCallback(acc, callbackExpectOk);
+    redisClusterAsyncSetDisconnectCallback(acc, callbackExpectOk);
+    redisClusterSetOptionAddNodes(acc->cc, CLUSTER_NODE);
+    redisClusterSetOptionUsername(acc->cc, CLUSTER_USERNAME);
+    redisClusterSetOptionPassword(acc->cc, CLUSTER_PASSWORD);
+
+    struct event_base *base = event_base_new();
+    redisClusterLibeventAttach(acc, base);
+
+    int ret;
+    ret = redisClusterConnect2(acc->cc);
+    assert(ret == REDIS_OK);
+    assert(acc->err == 0);
+    assert(acc->cc->err == 0);
+
+    // Test connection
+    ExpectedResult r = {
+        .type = REDIS_REPLY_STATUS, .str = "OK", .disconnect = true};
+    ret = redisClusterAsyncCommand(acc, commandCallback, &r, "SET key1 Hello");
+    assert(ret == REDIS_OK);
+
+    event_base_dispatch(base);
+
+    redisClusterAsyncFree(acc);
+    event_base_free(base);
+}
+
 int main() {
 
     test_password_ok();
     test_password_wrong();
     test_password_missing();
     test_multicluster();
+    test_username_ok();
+    test_username_disabled();
```

Comment:
Better put the username tests together with the password tests (before multicluster). They're related.

Reply:
Done